### PR TITLE
 1105 add graphQL schema file creation action

### PR DESCRIPTION
**Pull request by SilinMykola**
**Description** (*)
Added ability to create a `schema.graphqls` file from the action menu.
image

**Fixed Issues (if relevant)**
1. Fixes magento/magento2-phpstorm-plugin#1105

**Questions or comments**

**Contribution checklist** (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with integration/functional tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: src/com/magento/idea/magento2plugin/actions/generation/generator/GraphQLSchemaGenerator.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiFile;
+import com.magento.idea.magento2plugin.actions.generation.generator.util.FindOrCreateGraphQLSchema;
+import java.util.Properties;
+
+public class GraphQLSchemaGenerator extends FileGenerator {
+
+    private final String moduleName;
+
```
Comment (bohdan-harniuk) [inline src/com/magento/idea/magento2plugin/actions/generation/generator/GraphQLSchemaGenerator.java:16]: Please, remove unnecessary empty line.

File: src/com/magento/idea/magento2plugin/actions/generation/generator/GraphQLSchemaGenerator.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    /**
+     * Constructor for graphQL schema file generator.
+     *
+     * @param moduleName String
+     * @param project String
+     */
+    public GraphQLSchemaGenerator(
+            final String moduleName,
+            final Project project) {
```
Comment (bohdan-harniuk) [inline src/com/magento/idea/magento2plugin/actions/generation/generator/GraphQLSchemaGenerator.java:28]: Please, reformat constructor:
```suggestion
            final Project project
) {
```

File: src/com/magento/idea/magento2plugin/magento/files/GraphQLSchema.java [hunk 3]
```diff
@@ -2,8 +2,35 @@
...
  */
+
 package com.magento.idea.magento2plugin.magento.files;
 
-public class GraphQLSchema {
-    public static String FILE_NAME = "schema.graphqls";
+import com.intellij.lang.Language;
+import com.jetbrains.php.lang.PhpLanguage;
+
+public class GraphQLSchema implements ModuleFileInterface {
```
Comment (bohdan-harniuk) [inline src/com/magento/idea/magento2plugin/magento/files/GraphQLSchema.java:11]: All file models should have suffix **File**

File: src/com/magento/idea/magento2plugin/magento/files/GraphQLSchema.java [hunk 4]
```diff
@@ -2,8 +2,35 @@
...
+    }
+
+    @Override
+    public String getTemplate() {
+        return TEMPLATE;
+    }
+
+    @Override
+    public Language getLanguage() {
+        return PhpLanguage.INSTANCE;
```
Comment (bohdan-harniuk) [inline src/com/magento/idea/magento2plugin/magento/files/GraphQLSchema.java:34]: Why did you use PhpLanguage.INSTANCE?

File: src/com/magento/idea/magento2plugin/actions/generation/generator/GraphQLSchemaGenerator.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+ */
+
+package com.magento.idea.magento2plugin.actions.generation.generator;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiFile;
+import com.magento.idea.magento2plugin.actions.generation.generator.util.FindOrCreateGraphQLSchema;
+import java.util.Properties;
+
+public class GraphQLSchemaGenerator extends FileGenerator {
```
Comment (bohdan-harniuk) [inline src/com/magento/idea/magento2plugin/actions/generation/generator/GraphQLSchemaGenerator.java:13]: Where did you use this generator?

**Review by bohdan-harniuk — CHANGES_REQUESTED** [review]
Hello, @SilinMykola!

Thank you for your work!
Please, review my comment.

Regards,

File: src/com/magento/idea/magento2plugin/magento/files/GraphQLSchema.java (hunk 3 above)
Comment (SilinMykola) [reply src/com/magento/idea/magento2plugin/magento/files/GraphQLSchema.java:11]: fixed

File: src/com/magento/idea/magento2plugin/magento/files/GraphQLSchema.java (hunk 4 above)
Comment (SilinMykola) [reply src/com/magento/idea/magento2plugin/magento/files/GraphQLSchema.java:34]: fixed

File: src/com/magento/idea/magento2plugin/actions/generation/generator/GraphQLSchemaGenerator.java (hunk 5 above)
Comment (SilinMykola) [reply src/com/magento/idea/magento2plugin/actions/generation/generator/GraphQLSchemaGenerator.java:13]: @bohdan-harniuk Hi! Thanks for your review. I fixed. Please, check. Thanks!

**Review by bohdan-harniuk — APPROVED** [review]
Great work! Thank you!

**Pull request merged**